docs/release-notes/flink-1.11.md

```diff
@@ -0,0 +1,291 @@
+Starting from Flink 1.11 the `StateBackendFactory#createFromConfig` interface now takes `ReadableConfig` instead of `Configuration`.
+A `Configuration` class is still a valid argument to that method, as it implements the ReadableConfig interface.
+Implementors of custom `StateBackend` should adjust their implementations.
+
+#### Removal of deprecated OptionsFactory and ConfigurableOptionsFactory classes ([FLINK-18242](https://issues.apache.org/jira/browse/FLINK-18242))
+The deprecated `OptionsFactory` and `ConfigurableOptionsFactory` classes have been removed.
+Please use `RocksDBOptionsFactory` and `ConfigurableRocksDBOptionsFactory` instead.
+Please also recompile your application codes if any class extends `DefaultConfigurableOptionsFactory`.
+
+#### Enabled by default setTotalOrderSeek ([FLINK-17800](https://issues.apache.org/jira/browse/FLINK-17800))
+Since Flink-1.11 the option `setTotalOrderSeek` will be enabled by default for RocksDB's `ReadOptions`.
+This is in order to prevent user from miss using `optimizeForPointLookup`.
+For backward compatibility we support customizing `ReadOptions` through `RocksDBOptionsFactory`.
+Please set `setTotalOrderSeek` back to false if any performance regression observed (it shouldn't happen according to our testing).
+
+#### Increased default size of `state.backend.fs.memory-threshold` ([FLINK-17865](https://issues.apache.org/jira/browse/FLINK-17865))
+The default value of `state.backend.fs.memory-threshold` has been increased from 1K to 20K to prevent too many small files created on remote FS for small states.
+Jobs with large parallelism on source or stateful operators may have "JM OOM" or "RPC message exceeding maximum frame size" problem with this change.
+If you encounter such issues please manually set the configuration back to 1K.
+
+### PyFlink
+#### Throw exceptions for the unsupported data types ([FLINK-16606](https://issues.apache.org/jira/browse/FLINK-16606))
+DataTypes can be configured with some parameters, e.g., precision.
+However in previous releases, the precision provided by users was not taking any effect and default value for the precision was being used.
+To avoid confusion since Flink 1.11 exceptions will be thrown if the value is not supported to make it more visible to users. 
+Changes include:
+- the precision for `TimeType` could only be `0`
+- the length for `VarBinaryType`/`VarCharType` could only be `0x7fffffff`
+- the precision/scale for `DecimalType` could only be `38`/`18`
+- the precision for `TimestampType`/`LocalZonedTimestampType` could only be `3`
+- the resolution for `DayTimeIntervalType` could only be `SECOND` and the `fractionalPrecision` could only be `3`
+- the resolution for `YearMonthIntervalType` could only be `MONTH` and the `yearPrecision` could only be `2`
+- the `CharType`/`BinaryType`/`ZonedTimestampType` is not supported
```

Comment:
I believe it's this:

>Or does it mean that the only valid precision value for TimeType is 0 and if one configures a different value, then it will fail with an exception?

Yes, of course it should be `can` instead of `could`. I will rephrase it.